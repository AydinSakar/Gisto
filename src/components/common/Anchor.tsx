import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

import { IAnchor } from 'types/Interfaces.d';

const Link = styled.a`
  cursor: pointer;
  color: ${(props) => (props.color ? props.color : props.theme.baseAppColor)};
  text-decoration: none;

  :hover {
    color: ${(props) => props.theme.headerBgLightest};
  }
`;

const Anchor = ({ href, onClick, children, download, color, className }: IAnchor) => (
  <Link href={href} className={className} color={color} download={download} onClick={onClick}>
    {children}
  </Link>
);

Anchor.propTypes = {
  href: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node,
  download: PropTypes.string,
  color: PropTypes.string,
  className: PropTypes.string
};

export default Anchor;
