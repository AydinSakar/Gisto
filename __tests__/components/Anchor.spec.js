import Anchor from 'components/common/Anchor';

test('render Anchor', () => {
  const wrapper = shallow(<Anchor download="file.zip" href="https://github.com"/>);

  expect(wrapper).toMatchSnapshot();
});
