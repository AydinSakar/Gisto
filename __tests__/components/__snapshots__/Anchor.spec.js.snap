// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Anchor 1`] = `
<styled.a
  download="file.zip"
  href="https://github.com"
/>
`;
